settings.mk

```
@@ -180,6 +180,14 @@ endif
 
 TESTNG=$(LIB_DIR)$(D)testng.jar$(P)$(LIB_DIR)$(D)jcommander.jar
 RESOURCES_DIR=$(JVM_TEST_ROOT)$(D)TestConfig$(D)resources
+ifdef JTREG_JAR
+    $(ECHO) "Custom jtreg jar declared: $(JTREG_JAR)"
+   CUSTOM_JTREG_JAR=true
+else
+   CUSTOM_JTREG_JAR=false
+   JTREG_JAR=$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar
+endif
+
```

```suggestion
ifdef CUSTOM_JTREG_JAR
    $(ECHO) "Custom jtreg jar declared: $(CUSTOM_JTREG_JAR)"
   JTREG_JAR=$(CUSTOM_JTREG_JAR)
   # IS_CUSTOM_JTREG=true
else
   JTREG_JAR=$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar
   # IS_CUSTOM_JTREG=false
endif

```

I think the variable naming you chose will confuse people, so suggesting the above changes, since there _can_ always be a JTREG_JAR, but in some cases it will be overridden with a custom value. 

I would expect CUSTOM_JTREG_JAR to contain the path to a custom jtreg jar. If it is still needed, I would expect a boolean variable to be named something like IS_CUSTOM_JTREG or OVERRIDE_WITH_CUSTOM_JTREG.  Not actually sure you need it (maybe if later you want to query if a custom jtreg jar is in use).

---

Sounds reasoanble. the IS_CUSTOM_JTREG will become obsoleted as it will be easy to check  CUSTOM_JTREG_JAR. I will include it to rework due to missing `$(TEST_RESROOT)$(D)jtreg`